src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -183,15 +183,29 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
             IsRemote = !(uri?.IsFile ?? true);
         }
 
-       public string GetCompletePath() {
+        public void PathEditCompleted() {
+            // After the user is done editing the path, we fix it up automatically
+            Path = GetCompletePath();
+
+            // Conveniently populate the name, if the user directly edited the path
+            // without specifying a name.
+            if (string.IsNullOrWhiteSpace(Name)) {
```

Comment:
Actually, maybe it's better to split this away from here, and instead handle it in TextChanged event (or whatever it's called), rather than on LostFocus. That way, you'll see the name update (if it was empty or auto-generated) directly as you're typing, which I think will look nicer, and it should also be easier to get the old property value in that event.